src/OpenLoco/Utility/Numeric.cpp

```diff
@@ -39,8 +39,8 @@ namespace OpenLoco::Utility
         uint8_t success = _BitScanReverse(&i, source);
         return success != 0 ? i : -1;
 #elif defined(__GNUC__)
-        int32_t success = __builtin_ffs(__builtin_bswap32(source));
-        return success - 1;
+        int32_t success = __builtin_clz(source) ^ 31;
```

Comment:
isn't __builtin_clz undefined when source is 0?
So you need to do a branch:
```
success = source == 0 ? -1 : __builtin_clz(source) ^ 31;
```

Reply:
Sure, I didn't know whether we expected this to be called for zero values.

Reply:
I think it is used with 0 input. I'm sure bsf is used with input 0 so i can assume bsr would also end up being used with input 0